fe/fe-core/src/main/java/org/apache/doris/persist/VisitPartitionsInfo.java

```diff
@@ -0,0 +1,89 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.persist;
+
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.persist.gson.GsonUtils;
+import com.google.gson.annotations.SerializedName;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * used for batch store partitions visit info when a query arrive.
+ */
+
+public class VisitPartitionsInfo implements Writable {
+    @SerializedName(value = "dbId")
+    private long dbId;
+    @SerializedName(value = "tableId")
+    private long tableId;
+    @SerializedName(value = "partitionIds")
+    private Collection<Long> partitionIds;
+    @SerializedName(value = "timestamp")
+    private long timestamp;
+
+    public VisitPartitionsInfo(long dbId, long tableId, Collection<Long> partitionIds, long timestamp) {
+        this.dbId = dbId;
+        this.tableId = tableId;
+        this.partitionIds = partitionIds;
+        this.timestamp = timestamp;
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        Text.writeString(out, GsonUtils.GSON.toJson(this));
+    }
+
+    public static VisitPartitionsInfo read(DataInput in) throws IOException {
+        String json = Text.readString(in);
+        return GsonUtils.GSON.fromJson(json, VisitPartitionsInfo.class);
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
we need to override hashCode when we override equals

Reply:
Actually this function is never used, I'll delete it.